src/emitter.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Mixin that injects the events API into its host.
+ *
+ * @class Emitter
+ * @static
+ */
+
+CKEDITOR.define( [ 'eventinfo', 'utils' ], function( EventInfo, utils ) {
+	var EmmitterMixin = {
+		/**
+		 * Registers a callback function to be executed when an event is fired.
+		 *
+		 * @param {String} event The name of the event.
+		 * @param {Function} callback The function to be called on event.
+		 * @param {Object} [ctx] The object that represents `this` in the callback. Defaults to the object firing the
+		 * event.
+		 * @param {Number} [priority] The priority of this callback in relation to other callbacks to that same event.
+		 * Lower values are called first. Defaults to `10`.
+		 */
+		on: function( event, callback, ctx, priority ) {
+			var callbacks = getCallbacks( this, event );
+
+			var wasAdded;
+
+			// Priority defaults to 10.
+			if ( typeof priority != 'number' ) {
```

Comment:
I was wondering if this is worth for such kind of basic checks.

Btw, we have isObject() and isFunction() not because we need them in general, just because utils itself needs it.

In fact, we're suppose to remove such functions from utils from now with #7. I'm just waiting for #4 to get merged to propose this change there (it's already pushed in t/7).